app/views/application/dashboard.html.erb

```diff
@@ -0,0 +1,79 @@
+<div class="container">
+  <div class="dash col-md-6">
+    <div class="dash">
+      <h1> daily dose </h1>
+      <span class="quote">
+        <%= @quotes.sample %>
+      </span>
+    </div>
+    <div class="dash">
+      <h1> sort by tag </h1>
+      <div class="tabs">
+        <% @tags.each do |tag| %>
+          <% if !params[:extra].nil? && params[:extra][:tag_ids] %>
+            <% if tag.id == params[:extra][:tag_ids].to_i %>
+              <%= link_to tag.title, dashboard_path(:extra => { :tag_ids => tag.id }), {:class=>"selected_tab"} %> 
+            <% else %>
+              <%= link_to tag.title, dashboard_path(:extra => { :tag_ids => tag.id }), {:class=>"normal_tab"} %> 
+            <% end %>
+          <% else %>
+            <%= link_to tag.title, dashboard_path(:extra => { :tag_ids => tag.id }), {:class=>"normal_tab"} %> 
+          <% end %>
+        <% end %>
+      </div>
+    </div>
+    <div class="dash">
+      <h1> sort by location </h1>
+      <div class="tabs">
+        <% @locations.sort.each do |location| %>
+          <% if !params[:extra].nil? && params[:extra][:location] %>
+            <% if location == params[:extra][:location] %>
+              <%= link_to location, dashboard_path(:extra => {:location => location}), {:class=>"selected_tab"}  %>  
+            <% else %>
+              <%= link_to location, dashboard_path(:extra => {:location => location}), {:class=>"normal_tab"} %>  
+            <% end %>
+          <% else %>
+            <%= link_to location, dashboard_path(:extra => {:location => location}), {:class=>"normal_tab"} %>  
+          <% end %>
+        <% end %>
+      </div>
+    </div>
+    <div class="dash">
+      <h1> sort by interest </h1>
+      <div class="tabs">
+        <% @interest_levels.each do |range| %>
+          <% if !params[:extra].nil? && params[:extra][:interest] %>
+            <% if range == params[:extra][:interest] %>
+              <%= link_to range, dashboard_path(:extra => {:interest => range}), {:class=>"selected_tab"} %>
+            <% else %>
+              <%= link_to range, dashboard_path(:extra => {:interest => range}), {:class=>"normal_tab"} %>
+            <% end %>
+          <% else %>
+            <%= link_to range, dashboard_path(:extra => {:interest => range}), {:class=>"normal_tab"} %>
+          <% end %>
+        <% end %> 
+      </div>
+    </div>
+    <div class="dash">
+      <h1> 3 most interesting companies </h1>
+      <% @companies_by_max_interest.each do |k, v| %>
```

Comment:
i encourage you to choose better names than `k` and `v` - these aren't very descriptive